aten/src/ATen/cpu/vec/vec256/vec256_double.h

```diff
@@ -100,6 +100,10 @@ template <> class Vectorized<double> {
   Vectorized<double> isnan() const {
     return _mm256_cmp_pd(values, _mm256_set1_pd(0.0), _CMP_UNORD_Q);
   }
+  bool has_infinite() const {
+    __m256d self_sub  = _mm256_sub_pd(values, values);
+    return (_mm256_movemask_epi8(_mm256_castpd_si256(self_sub)) & 0x77777777) != 0;
```

Comment:
Can we compare with a vector of INFINITY for equality directly?
```c++
__mmask8 mask;
__m256d inf = _mm256_set1_pd(INFINITY);
mask = _mm256_cmp_pd_mask(values, inf, _CMP_EQ_OQ); // compare v and inf for equality
return mask != 0;
```

Reply:
Renamed the function. Check nan and inf at once.

Reply:
I still don't follow why the code you provide can work. Can you explain? I guess we'd better do comparison for nan and for inf separately with `_mm256_cmp_pd`.

Reply:
Subtraction from itself will pruduce +/-0.0 for finite inputs, else NaN and then using _mm256_movemask_epi8 to take one bit from each byte, including one from the exponent which will be non-zero for NaNs. movemask &  0x77777777  ignores the sign bit. Of course, we can do comparison for nan and for inf separately.

Reply:
OK. Sounds reasonable.